transaction/transaction.go

```diff
@@ -356,6 +357,23 @@ func (tx *Transaction) TxHash() chainhash.Hash {
 	return chainhash.DoubleHashH(buf)
 }
 
+// IssuanceHash generates the Hash for the issuance transaction.
+func (tx *Transaction) IssuanceHash() (chainhash.Hash, error) {
+	hasIssuance := false
+	for _, v := range tx.Inputs {
+		if v.Issuance != nil {
+			hasIssuance = true
+		}
+	}
+	if !hasIssuance {
+		return [32]byte{}, errors.New("no input with issuance")
+	}
+
+	// Encode the transaction and calculate double sha256 on the result.
+	buf, _ := tx.serialize(nil, false, false, false)
```

Comment:
can you explain  why all false?

Reply:
While debugging I noticed that when serialising, [this](https://github.com/sekulicd/go-elements/blob/4e90ba16eaa7ba396562acb763f57c123a33f678/transaction/transaction.go#L731) is not reached, cause 'zeroFlag' was true in TxHash() method, after I changed this flag to false, txHash was as expected.
Maybe @altafan can explain better what this flag is doing and if he has other idea on how to solve this.